File: php/examples/publish-example.php
<?php

    ## Capture Publish and Subscribe Keys from Command Line
    $publish_key   = isset($argv[1]) ? $argv[1] : false;
    $subscribe_key = isset($argv[2]) ? $argv[2] : false;
    $message       = isset($argv[3]) ? $argv[3] : false;

    ## Print usage if missing info.
    if (!($publish_key && $subscribe_key && $message)) {
echo("
    ==============
    EXAMPLE USAGE:
    ==============
    php ./publish-example.php PUBLISH-KEY SUBSCRIBE-KEY 'ANY MESSAGE HERE'
    php ./publish-example.php PUBLISH-KEY SUBSCRIBE-KEY 'Hey what is up?'

");
        exit();
    }

    ## Require Pubnub API
    echo("Loading Pubnub.php Class\n");
    require('../Pubnub.php');

    ## -----------------------------------------
    ## Create Pubnub Client API (INITIALIZATION)
    ## -----------------------------------------
    echo("Creating new Pubnub Client API\n");
    $pubnub = new Pubnub( $publish_key, $subscribe_key );

    ## ----------------------
    ## Send Message (PUBLISH)
    ## ----------------------
    echo("Sending a message with Publish Function\n");
    $info = $pubnub->publish(array(
        'channel' => 'hello_world2',
        'message' => $message
    ));

    echo("Response:\n");
    var_dump($info);

?>

File: ruby/examples/publish-example.rb
## -----------------------------------
## PubNub Ruby API Publish Example
## -----------------------------------

## including required libraries
require './lib/pubnub.rb'
require 'json'

## declaring publish_key, subscribe_key, secret_key, cipher_key, channel, ssl flag, messages
publish_key   = 'demo'
subscribe_key = 'demo'
secret_key    = 'demo'
cipher_key    = 'demo'
ssl_on        = false
channel       = 'hello_world'

strMessage = "Hello World"
arrMessage = ["Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday"]
objMessage = {"Name"=>"John","Age"=>"23"}

## Print usage if missing info.
if !strMessage && arrMessage && objMessage
  puts('
  Get API Keys at http://www.pubnub.com/account
  ==============
  EXAMPLE USAGE:
  ==============
  ruby publish-example.rb PUB-KEY SUB-KEY SECRET-KEY "message text" SSL-ON
  ruby publish-example.rb demo demo "demo" "hey what is up?" true
  ')
  exit()
end

## Pubnub state initialization (INITIALIZATION)
puts('Initialize new Pubnub state')
pubnub = Pubnub.new(publish_key,subscribe_key,secret_key,cipher_key,ssl_on)

## Send Message (PUBLISH) -- String
puts("\nSending message in String format")
info = pubnub.publish({'channel' => channel ,'message' => strMessage})
puts(info)

## Send Message (PUBLISH) -- Array
puts("\nSending message in Array format")
info = pubnub.publish({'channel' => channel ,'message' => arrMessage})
puts(info)

## Send Message (PUBLISH) -- Object (Dictionary)
puts("\nSending message in Dictionary Object format")
info = pubnub.publish({'channel' => channel ,'message' => objMessage})
puts(info)
